content/docs/esc/integrations/external-secrets-operator.md

```diff
@@ -0,0 +1,489 @@
+
+#### Select your Pulumi supported language
+
+Create a new Pulumi program in your preferred language and add the following code to deploy the External Secrets Operator.
+
+```bash
+pulumi new <your-preferred-language> --name external-secrets-operator
+```
+
+Add the following code to the `index.ts`, `index.py`, or `main.go` file:
+
+{{< chooser language "typescript,python,go" />}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as kubernetes from "@pulumi/kubernetes";
+
+const config = new pulumi.Config();
+
+const externalSecretsNamespace = new kubernetes.core.v1.Namespace("external-secrets-namespace", {
+    metadata: {
+        name: "external-secrets",
+    }
+});
+const externalSecretsRelease = new kubernetes.helm.v3.Release("external-secrets-release", {
+    chart: "external-secrets",
+    version: "0.10.4",
+    namespace: externalSecretsNamespace.metadata.apply(metadata => metadata.name),
+    repositoryOpts: {
+        repo: "https://charts.external-secrets.io",
+    },
+});
+const patSecret = new kubernetes.core.v1.Secret("patSecret", {
+    metadata: {
+        namespace: externalSecretsNamespace.metadata.apply(metadata => metadata.name),
+        name: "pulumi-access-token",
+    },
+    stringData: {
+        PULUMI_ACCESS_TOKEN: config.require("pulumi-pat"),
+    },
+    type: "Opaque",
+});
+
+const clusterSecretStore = new kubernetes.apiextensions.CustomResource("cluster-secret-store", {
+    apiVersion: "external-secrets.io/v1beta1",
+    kind: "ClusterSecretStore",
+    metadata: {
+        name: "secret-store",
+    },
+    spec: {
+        provider: {
+            pulumi: {
+                organization: pulumi.getOrganization(),
+                project: "dirien",
+                environment: "hello-world",
+                accessToken: {
+                    secretRef: {
+                        name: patSecret.metadata.name,
+                        key: "PULUMI_ACCESS_TOKEN",
+                        namespace: patSecret.metadata.namespace,
+                    },
+                },
+            },
+        },
+    },
+}, { dependsOn: externalSecretsRelease });
+
+const externalSecret = new kubernetes.apiextensions.CustomResource("external-secret", {
+    apiVersion: "external-secrets.io/v1beta1",
+    kind: "ExternalSecret",
+    metadata: {
+        name: "secret",
+        namespace: "default",
+    },
+    spec: {
+        data: [
+            {
+                secretKey: "esc-secret",
+                remoteRef: {
+                    key: "hello",
+                }
+            }
+        ],
+        refreshInterval: "20s",
+        secretStoreRef: {
+            kind: clusterSecretStore.kind,
+            name: clusterSecretStore.metadata.name,
+        }
+    },
+}, { dependsOn: externalSecretsRelease });
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
```

Comment:
Mind running this code through `black`? We usually left-justify Python code to make it easier to read: 

![image](https://github.com/user-attachments/assets/276deacc-18cb-464e-9ea0-d43ebc57d16d)